packages/devextreme/js/__internal/ui/map/m_provider.dynamic.azure.ts

```diff
@@ -1,50 +1,57 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
 /* eslint-disable @typescript-eslint/no-misused-promises */
 import Color from '@js/color';
 import $ from '@js/core/renderer';
 import ajax from '@js/core/utils/ajax';
 import { noop } from '@js/core/utils/common';
-import { map } from '@js/core/utils/iterator';
 import { isDefined } from '@js/core/utils/type';
 import { getWindow } from '@js/core/utils/window';
+import type { MapType, RouteMode } from '@js/ui/map';
 import errors from '@js/ui/widget/ui.errors';
 
+import type {
+  LocationOption,
+  MarkerObject, MarkerOptions, RouteObject, RouteOptions,
+} from './m_provider.dynamic';
 import DynamicProvider from './m_provider.dynamic';
 
 const window = getWindow();
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
 declare let atlas: any;
 
 const AZURE_BASE_LINK = 'https://atlas.microsoft.com/';
 let AZURE_JS_URL = `${AZURE_BASE_LINK}sdk/javascript/mapcontrol/3/atlas.min.js`;
 let AZURE_CSS_URL = `${AZURE_BASE_LINK}/sdk/javascript/mapcontrol/3/atlas.min.css`;
+// eslint-disable-next-line @typescript-eslint/init-declarations
 let CUSTOM_URL;
 
 const MAP_MARKER_TOOLTIP_CLASS = 'dx-map-marker-tooltip';
 
 const CAMERA_PADDING = 50;
+// @ts-expect-error ts-error
+const azureMapsLoaded = (): boolean => Boolean(window.atlas?.Map);
 
-const azureMapsLoaded = function () {
-  // @ts-expect-error
-  return window.atlas?.Map;
-};
+export type AzureLocation = [number, number];
 
+// eslint-disable-next-line @typescript-eslint/init-declarations
 let azureMapsLoader;
 class AzureProvider extends DynamicProvider {
   _preventZoomChangeEvent?: boolean;
 
-  _mapReadyPromise?: Promise<void>;
+  _mapReadyPromise!: Promise<void>;
 
-  _mapType(type) {
+  _mapType(type: MapType): string {
     const mapTypes = {
       roadmap: 'road',
       satellite: 'satellite',
       hybrid: 'satellite_road_labels',
     };
-    return mapTypes[type] || mapTypes.roadmap;
+    return mapTypes[type] ?? mapTypes.roadmap;
   }
 
-  _movementMode(type) {
-    const movementTypes = {
+  _movementMode(type: RouteMode | string = ''): string {
```

Comment:
Default value should be: 
```suggestion
  _movementMode(type: RouteMode | string = movementTypes.driving): string {
```

And early return above can be removed

Reply:
no, i'd prefer keep it this way since even though I used RouteMode | string type here, it still possible someone pass smth like null which won't be handled in args default value